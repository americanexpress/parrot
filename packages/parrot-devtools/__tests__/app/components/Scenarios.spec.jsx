/*
 * Copyright 2019 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { shallow } from 'enzyme';

import ScenariosMiddleware from '../../../src/app/components/Scenarios';

const scenario = 'api-fetch';
const setScenario = jest.fn(() => Promise.resolve());
const scenariosData = {
  scenario,
  setScenario,
  loading: false,
  filteredScenarios: [{ name: scenario }, { name: 'scenario-2' }],
  loadScenarios: jest.fn(() => Promise.resolve()),
};
const mockProps = { scenariosData };

describe('Scenarios', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('ScenariosMiddleware', () => {
    it('should render scenarios middleware component', () => {
      const result = shallow(<ScenariosMiddleware {...mockProps} />);

      expect(result).toMatchSnapshot();

      result
        .find('button')
        .first()
        .simulate('click');

      expect(setScenario).toHaveBeenCalledWith(scenario);
    });

    it('should loading state', () => {
      const props = {
        scenariosData: {
          ...scenariosData,
          loading: true,
        },
      };
      const result = shallow(<ScenariosMiddleware {...props} />);

      expect(result).toMatchSnapshot();
    });

    it('should inform the user of failure', () => {
      const props = {
        scenariosData: {
          ...scenariosData,
          loading: false,
          scenario: '',
          scenarios: [],
        },
      };
      const result = shallow(<ScenariosMiddleware {...props} />);

      expect(result).toMatchSnapshot();
    });
  });
});
