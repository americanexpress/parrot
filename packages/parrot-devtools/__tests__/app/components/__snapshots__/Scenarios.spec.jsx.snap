// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scenarios ScenariosMiddleware should inform the user of failure 1`] = `
<styled.div>
  <div
    className="alert alert-warn alert-dialog alert-lg"
    role="alert"
  >
    <span
      className="alert-icon dls-icon-warning icon-lg"
    />
    <h1
      className="heading-4 alert-header"
    >
      Failed to Load Scenarios
    </h1>
    <p
      className="alert-subtitle"
    >
      <span>
        Try 
      </span>
      <button
        onClick={[MockFunction]}
        type="button"
      >
        refreshing
      </button>
      <span>
         or updating the middleware URL in settings.
      </span>
    </p>
  </div>
</styled.div>
`;

exports[`Scenarios ScenariosMiddleware should loading state 1`] = `
<styled.div>
  <div
    className="progress-circle progress-indeterminate margin-auto"
  />
</styled.div>
`;

exports[`Scenarios ScenariosMiddleware should render scenarios middleware component 1`] = `
<styled.div
  className="border"
>
  <ul
    className="nav-menu"
  >
    <li
      className="nav-item"
      key="api-fetch"
    >
      <button
        aria-pressed={true}
        className="nav-link btn-secondary text-align-left fluid "
        onClick={[Function]}
        type="button"
      >
        api-fetch
      </button>
    </li>
    <li
      className="nav-item"
      key="scenario-2"
    >
      <button
        aria-pressed={false}
        className="nav-link btn-secondary text-align-left fluid border-0"
        onClick={[Function]}
        type="button"
      >
        scenario-2
      </button>
    </li>
  </ul>
</styled.div>
`;
